tests/Api/ProductMediaFileApiIntegration.php

```diff
@@ -0,0 +1,214 @@
+        $mediaFiles[1] = $this->sanitizeMediaFile($mediaFiles[1]);
+
+        $this->assertSameContent($expectedSecondMediaFile, $mediaFiles[1]);
+
+        $secondPage = $firstPage->getNextPage();
+        $this->assertInstanceOf('\Akeneo\Pim\Pagination\PageInterface', $secondPage);
+        $this->assertSame($uriGenerator->generate(ProductMediaFileApi::MEDIA_FILES_PATH, [], [
+            'page'       => 1,
+            'limit'      => 2,
+            'with_count' => false,
+        ]), $secondPage->getPreviousLink());
+
+        $lastPage = $secondPage->getNextPage();
+        $this->assertInstanceOf('\Akeneo\Pim\Pagination\PageInterface', $lastPage);
+        $this->assertNull($lastPage->getNextLink());
+        $this->assertNull($lastPage->getNextPage());
+
+        $this->assertCount(0, $lastPage->getItems());
+    }
+
+    public function testListPerPageWithCount()
+    {
+        $api = $this->createClient()->getProductMediaFileApi();
+
+        $firstPage = $api->listPerPage(2, true);
+        $this->assertInstanceOf('\Akeneo\Pim\Pagination\PageInterface', $firstPage);
+        $this->assertSame(4, $firstPage->getCount());
+    }
+
+    public function testAll()
+    {
+        $api = $this->createClient()->getProductMediaFileApi();
+        $mediaFiles = $api->all();
+
+        $this->assertInstanceOf('\Akeneo\Pim\Pagination\ResourceCursorInterface', $mediaFiles);
+
+        $mediaFilesCount = 0;
+        foreach ($mediaFiles as $mediaFile) {
+            $this->assertInternalType('array', $mediaFile);
+            $mediaFilesCount++;
+        }
+
+        $this->assertSame(4, $mediaFilesCount);
+    }
+
+    public function testGet()
+    {
+        $api = $this->createClient()->getProductMediaFileApi();
+        $code = $api->listPerPage(1)->getItems()[0]['code'];
+
+        $mediaFile = $api->get($code);
+        $this->assertInternalType('array', $mediaFile);
+
+        $expectedMediaFile = [
+            'code'              => $code,
+            'original_filename' => 'Ziggy-certification.jpg',
+            'mime_type'         => 'image/jpeg',
+            'size'              => 10513,
+            'extension'         => 'jpg',
+            '_links' => [
+                'download' => [
+                    'href' => $this->getUriGenerator()->generate(ProductMediaFileApi::MEDIA_FILE_DOWNLOAD_PATH, ['code' => $code]),
+                ],
+            ],
+        ];
+
+        $this->assertSameContent($expectedMediaFile, $mediaFile);
+    }
+
+    public function testCreate()
+    {
+        $api = $this->createClient()->getProductMediaFileApi();
+        $mediaFile = realpath(__DIR__.'/../fixtures/akeneo.png');
+
+        $response = $api->create($mediaFile, [
+            'identifier' => 'small_boot',
+            'attribute'  => 'side_view',
+            'scope'      => null,
+            'locale'     => null,
+        ]);
+
+        $this->assertSame(201, $response);
```

Comment:
That show us a problem in the design of this method! We don't get the generated media file code, so it's unusable for the user. It can be guess from the Location header (but need parsing of the URL though).